server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -156,9 +165,25 @@ private void handleUpdate(MailboxId mailboxId, MailboxUpdateRequest updateReques
         }
    }
 
-    private void checkRole(Optional<Role> role) throws SystemMailboxNotUpdatableException {
-        if (role.map(Role::isSystemRole).orElse(false)) {
-            throw new SystemMailboxNotUpdatableException();
+    private void assertNotSharedOutboxOrDraftMailbox(Mailbox mailbox, MailboxUpdateRequest updateRequest) {
+        Preconditions.checkArgument(!isSetSharedWithOn(updateRequest.getSharedWith(), mailbox, Role.OUTBOX), "Cannot share 'Outbox'");
+        Preconditions.checkArgument(!isSetSharedWithOn(updateRequest.getSharedWith(), mailbox, Role.DRAFTS), "Cannot share 'Draft'");
+    }
+
+    private boolean isSetSharedWithOn(Optional<Rights> rights, Mailbox mailbox, Role role) {
+        return mailbox.getRole().isPresent()
```

Comment:
return mailbox.hasRole(role) && rights.isPresent();

Reply:
And I would inline these conditions inside the preconditions to avoid that strange naming.